src/engine/source/builder/builders/opBuilderKVDB.cpp

```diff
@@ -154,23 +160,24 @@ base::Expression existanceCheck(const std::string& targetField,
     const auto name = formatHelperName(targetField, rawName, parameters);
 
     const auto dbName = parameters[0].m_value;
-
-    // Get DB
-    // TODO: Fix once KVDB is refactored
-    const auto result = kvdbManager->getHandler(dbName);
-    if (std::holds_alternative<base::Error>(result))
-    {
-        throw std::runtime_error(fmt::format("Engine KVDB builder: {}.", std::get<base::Error>(result).message));
-    }
-
     const std::string successTrace {fmt::format("[{}] -> Success", name)};
-
     const std::string failureTrace {
         fmt::format("[{}] -> Failure: Target field '{}' does not exist or it is not a string", name, targetField)};
 
+    auto resultHandler = kvdbScope->getKVDBHandler(dbName);
+
+    if (std::holds_alternative<base::Error>(resultHandler))
+    {
+        throw std::runtime_error(fmt::format("Engine KVDB builder: {}.", std::get<base::Error>(resultHandler).message));
+    }
+
     return base::Term<base::EngineOp>::create(
         name,
-        [=, kvdb = std::get<kvdb_manager::KVDBHandle>(result), targetField = std::move(targetField)](base::Event event)
+        [=, 
+        targetField = std::move(targetField),
+        kvdbHandler = std::move(std::get<std::shared_ptr<kvdbManager::IKVDBHandler>>(resultHandler))
+        ]
+        (base::Event event)
         {
             bool found = false;
             try // TODO We are only using try for JSON::get. Is correct to
```

Comment:
JSON::get only throws if the JPath is invalid, which should be checked on buildtime